src/edu/cmu/cs/sasylf/ast/Clause.java

```diff
@@ -55,19 +58,31 @@ public void add(Element e) {
 
 	protected List<Element> elements = new ArrayList<Element>();
 
-	@Override

```

Comment:
Having `implements Cloneable` in the declaration of `Clause` is redundant, because `Clause` implements `Element`, `Element` extends `Node`, `Node` implements `Span`, and `Span` implements `Cloneable`.

The reason we want to override `clone()` is to specialize the return type. If we don't override `clone()` in `Clause`, then the `clone()` that is nearest to `Clause` is the one inherited from `Object`, which returns `Object`. By overriding `clone()` in `Clause` we can cast the `Object` returned from `super.clone()` prior to returning it, which prevents us from needing to cast it outside of where `Clause.clone()` is called.